### PR TITLE
Complex forms and components UI

### PR DESCRIPTION
**Pull request by myieye**
Resolves most of #1089 

image
image
image

### CODE REVIEW

**Review by hahn-kev — CHANGES_REQUESTED** [review]
I made some changes as I went, feel free to adjust any of them as needed. I left 2 questions/comments.

File: frontend/viewer/src/lib/entry-editor/EntryOrSensePicker.svelte [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+              } else {
+                // let myself close
+                select();
+              }
+            }
+          }}>
+          <button slot="trigger" class="flex-1 flex justify-between items-center text-left max-w-full overflow-hidden"
+            on:click={() => {
+              if (disableExpand && !disabledEntry) {
+                // the change event above is not in use, so we need to do it here
```
[the remark this answers is not in the document]
Comment (myieye) [reply frontend/viewer/src/lib/entry-editor/EntryOrSensePicker.svelte:158]: The comment is valid in the context of the `if` it's inside.

[the remark this answers is not in the document]
Comment (myieye) [reply frontend/viewer/src/lib/entry-editor/EntryOrSensePicker.svelte:158]: Actually, only this part: `if (disableExpand)`.
The `disabledEntry` part should probably just set the disabled attribute of the button it's on.

File: frontend/viewer/src/lib/entry-editor/field-editors/ComplexFormTypes.svelte [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+>
+  <FieldTitle {id} {name} />
+
+  <div class="item-list-field">
+    <OrderedItemList bind:value {readonly} on:change={(e) => dispatch('change', { value })} getDisplayName={(type) => firstVal(type.name)}>
+      <svelte:fragment slot="actions">
+        <Button disabled on:click={() => openPicker = true} icon={mdiPlus} variant="fill-light" color="success" size="sm">
+          <div class="max-sm:hidden">Add Complex Form Type</div>
+        </Button>
+        <!-- TODO: implement form type picker or dropdown? <EntryOrSensePicker title="Add complex form" bind:open={openPicker} noSenses on:pick={(e) => addComplexFormType(e.detail)} /> -->
```
[the remark this answers is not in the document]
Comment (myieye) [reply frontend/viewer/src/lib/entry-editor/field-editors/ComplexFormTypes.svelte:49]: Yeah, I'd like to do that in a follow up PR.

File: frontend/viewer/src/lib/entry-editor/EntryOrSensePicker.svelte (hunk 1 above)
[the remark this answers is not in the document]
Comment (hahn-kev) [reply frontend/viewer/src/lib/entry-editor/EntryOrSensePicker.svelte:158]: I'm not sure I understand. By `change event` are you referring to this `on:click` handler?

[the remark this answers is not in the document]
Comment (myieye) [reply frontend/viewer/src/lib/entry-editor/EntryOrSensePicker.svelte]: I clarified the comment and did a tiny refactor 29446191bf06a2b73388378ada1427bb96e84bc4

**Review by hahn-kev — APPROVED** [review]

**Pull request merged**